lib/bcc_helper.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019 by Delphix. All rights reserved.
+ */
+
+/*
+ * This file contains helper utilities to write bcc scripts for delphix
+ * analytics and other bcc scripts.
+ */
+
+
+/* Structure used in average aggregations and zero initializer value*/
+typedef struct average {
+    u64 count;
+    u64 sum;
+} average_t;
+
+#define ZERO_AVERAGE {0, 0}
+
+
+/* define histogram key type with the aggregation key and a slot */
+#define HIST_KEY(hist_key_type, agg_key_type) \
+typedef struct { \
+     agg_key_type  agg_key; \
+     u64           slot; \
+} hist_key_type;
+
+/*
+ * This function returns the slot, or histogram bucket, for a value based
+ * on log linear distribution equivalent to dtrace llquantize(*, 10, 4, 10, 10).
+ * The maximum islot returned is 60 for values great than 10^10.
+ */
+static u64 log_lin_hist_slot(u64 value)
```

Comment:
Should we be using a similar C-style to Openzfs?

Reply:
Good idea.  I ran it through the cstyle in the zfs repo.